pkg/model/iam/iam_builder.go

```diff
@@ -275,6 +275,10 @@ func (r *NodeRoleMaster) BuildAWSPolicy(b *PolicyBuilder) (*Policy, error) {
 		addCiliumEniPermissions(p, resource, b.Cluster.Spec.IAM.Legacy)
 	}
 
+	if b.Cluster.Spec.Networking != nil && b.Cluster.Spec.Networking.Calico != nil && b.Cluster.Spec.Networking.Calico.AwsSrcDstCheck != "" {
+		addCalicoSrcDstCheckPermissions(p)
```

Comment:
Why do we add the permission regardless of the value of the "AwsSrcDstCheck" field? If it's "DoNothing," we wouldn't need this permission, right?

Reply:
Reading the documentation in file _calico.md_, it suggests that if "CrossSubnet" is true, then we behave as if "AwsSrcDstCheck" is set to "Disable," even if the latter field is not set at all. However, here, we're only adding this permission of "AwsSrcDstCheck" is nonempty. Did you intend to set the "AwsSrcDstCheck" field earlier? Is the documentation making the wrong promise, such that we do intend to force the user to set the "AwsSrcDstCheck" field?